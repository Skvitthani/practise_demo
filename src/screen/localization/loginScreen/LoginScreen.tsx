import i18n from '../../../i18n';
import React, {useState} from 'react';
import {useTranslation} from 'react-i18next';
import {NavigationType} from '../../../navigation';
import StringConst from '../../../utils/StringConst';
import ButtonComp from '../../../components/ButtonComp';
import InputTextComp from '../../../components/InputTextComp';
import LanguageModal from '../../../components/LanguageModal';
import {lang, setItemInAsync} from '../../../utils/helperFunction';
import {View, Text, StyleSheet, SafeAreaView} from 'react-native';
import {NativeStackScreenProps} from '@react-navigation/native-stack';

type LoginScreen = NativeStackScreenProps<NavigationType, 'LoginScreen'> & {
  navigation: () => void;
};

const LoginScreen: React.FC<LoginScreen> = ({navigation}) => {
  const [email, setEmail] = useState<string | number>('');
  const [password, setPassword] = useState<string | number>('');
  const [langModalVisible, setLangModalVisible] = useState(false);

  const {t} = useTranslation();

  return (
    <View style={styles.container}>
      <SafeAreaView />
      <InputTextComp
        inputTestID="name_input_text"
        value={email}
        placeholder={t(StringConst.Enter_Name)}
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setEmail(txt)}
      />
      <InputTextComp
        inputTestID="password_input_text"
        value={password}
        placeholder={t(StringConst.Enter_Password)}
        inputStyle={styles.inputStyle}
        onChangeText={(txt: string | number) => setPassword(txt)}
      />

      <View style={styles.loginBtn}>
        <ButtonComp title={t('Login')} />
      </View>

      <Text
        testID="create_account_screen_press"
        style={styles.createNewAccount}
        onPress={() => {
          navigation.navigate('CreateAccountScreen');
        }}>
        {t(StringConst.Create_New_Account)}
      </Text>
      <View style={styles.laugBtn}>
        <ButtonComp
          TitleTestId="select_language_button"
          title={t(StringConst.Select_Language)}
          onPress={() => {
            setLangModalVisible(!langModalVisible);
          }}
        />
      </View>
      <LanguageModal
        onApplyPress={selectedLang => {
          i18n.changeLanguage(selectedLang);
          setItemInAsync(lang, selectedLang);
          setLangModalVisible(!langModalVisible);
        }}
        modalVisible={langModalVisible}
        onRequestClose={() => {
          setLangModalVisible(!langModalVisible);
        }}
      />
    </View>
  );
};

export default LoginScreen;
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  title: {
    fontSize: 20,
    fontWeight: '800',
    color: '#000',
    marginTop: 100,
    alignSelf: 'center',
  },
  inputStyle: {
    paddingLeft: 20,
    height: 50,
    alignSelf: 'center',
    marginTop: 30,
    borderWidth: 0.5,
    borderRadius: 10,
    width: '90%',
  },
  loginBtn: {
    marginTop: 50,
  },
  laugBtn: {
    flex: 1,
    marginBottom: 50,
    justifyContent: 'flex-end',
  },

  createNewAccount: {
    fontSize: 18,
    fontWeight: '600',
    textDecorationLine: 'underline',
    marginTop: 50,
    alignSelf: 'center',
  },
});
