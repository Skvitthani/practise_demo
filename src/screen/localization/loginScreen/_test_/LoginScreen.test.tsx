import React from 'react';
import LoginScreen from '../LoginScreen';
import {fireEvent, render} from '@testing-library/react-native';

jest.mock('@react-native-async-storage/async-storage', () => ({
  getItem: jest.fn(),
}));

const component = (
  <LoginScreen
    navigation={{navigate: jest.fn()} as any}
    route={undefined as any}
  />
);

describe('LoginScreen', () => {
  test('renders correctly', () => {
    const {toJSON} = render(component);
    expect(toJSON()).toMatchSnapshot();
  });
  it('name text input', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('name_input_text');
    fireEvent.changeText(container);
  });
  it('password text input', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('password_input_text');
    fireEvent.changeText(container);
  });
  it('select language button', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('select_language_button');
    fireEvent.press(container);
  });
  it('create account screen press', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('create_account_screen_press');
    fireEvent.press(container);
  });
  it('select language', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('select_language_button');
    fireEvent.press(container);
  });
});
