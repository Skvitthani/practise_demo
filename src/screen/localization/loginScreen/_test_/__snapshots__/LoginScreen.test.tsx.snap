// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Name"
    placeholderTextColor="black"
    style={
      [
        {
          "backgroundColor": "#FCF5ED",
          "borderRadius": 15,
        },
        {
          "alignSelf": "center",
          "borderRadius": 10,
          "borderWidth": 0.5,
          "height": 50,
          "marginTop": 30,
          "paddingLeft": 20,
          "width": "90%",
        },
      ]
    }
    testID="name_input_text"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Password"
    placeholderTextColor="black"
    style={
      [
        {
          "backgroundColor": "#FCF5ED",
          "borderRadius": 15,
        },
        {
          "alignSelf": "center",
          "borderRadius": 10,
          "borderWidth": 0.5,
          "height": 50,
          "marginTop": 30,
          "paddingLeft": 20,
          "width": "90%",
        },
      ]
    }
    testID="password_input_text"
    value=""
  />
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#706233",
          "borderRadius": 20,
          "marginHorizontal": 30,
          "opacity": 1,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
  <Text
    onPress={[Function]}
    style={
      {
        "alignSelf": "center",
        "fontSize": 18,
        "fontWeight": "600",
        "marginTop": 50,
        "textDecorationLine": "underline",
      }
    }
    testID="create_account_screen_press"
  >
    Create New Account
  </Text>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 50,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#706233",
          "borderRadius": 20,
          "marginHorizontal": 30,
          "opacity": 1,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
        testID="select_language_button"
      >
        Select Language
      </Text>
    </View>
  </View>
</View>
`;
